src/emulator/storage/metadata.ts

```diff
@@ -180,7 +188,12 @@ export class StoredFileMetadata {
     }
 
     if (incoming.metadata) {
-      this.customMetadata = incoming.metadata;
+      // Convert all values to strings
+      this.customMetadata = this.customMetadata ? { ...this.customMetadata } : {};
+      for (const k of Object.keys(incoming.metadata)) {
+        const v = incoming.metadata[k];
+        this.customMetadata[k] = (v === null ? null : String(v)) as any;
```

Comment:
Is this `as any` cast necessary? If it is (i.e. we're putting `null` into an interface that only expects `string`, please update this to `as unknown as string`.

It's more verbose but it avoids using any, and makes clear _how_ we're circumventing the type-system